accepted/2024/trusted-publishers-oidc-for-nuget-push.md

```diff
@@ -0,0 +1,538 @@
+package registry (NuGet.org in this case) and thereafter depend on their secure build environment (such as GitHub
+Actions) to properly authenticate with NuGet.org, exercising the trust policy while publishing a package.
+
+Any secret related to this new flow is short-lived, meaning leaks are less impactful. The burden of rotating an API key
+in a secret store that is available to the package author's publishing pipeline is removed. 
+
+So, it's a usability win and a security win!
+
+## Functional explanation
+
+At the core of this design is a requirement. The package author must publish their package to NuGet.org from an
+environment that is supported by the Trusted Publishers feature. The build environment must support OpenID Connect
+(OIDC) tokens because they take the place of long-lived NuGet API keys as the credential. The NuGet package source must
+also be configured to allow OIDC tokens from this specific build environment.
+
+For the sake of understanding, we'll use GitHub and GitHub Actions as the example below, but it's important to note that
+other build environments (i.e. other Trusted Publishers) could be added later without too much difficulty. To think
+about Bitbucket or GitLab support (for example), simply replace the GitHub feature mentioned with the related feature on
+the other forge.
+
+A package author performs the following steps to make use of the Trusted Publishers feature:
+
+[Prerequisite](#prerequisite-set-up-a-github-repository-for-the-nuget-package): set up a GitHub repository for the NuGet
+package
+   - This is all the package's code to be built and uploaded in the trusted build environment.
+
+[Step 1](#step-1-add-a-github-actions-trust-policy-to-your-nugetorg-account): add a GitHub Actions trust policy to your
+NuGet.org account
+   - This is to instruct NuGet.org to accept a pattern of GitHub tokens as a valid credential, in lieu of API key.
+
+[Step 2](#step-2-add-a-github-actions-workflow-with-nugetorg-trusted-publisher-authentication): add a GitHub Actions
+workflow with NuGet.org Trusted Publisher authentication
+   - This instructs the NuGet client tooling to authenticate with NuGet.org using a GitHub Actions OIDC token.
+
+[Step 3](#step-3-execute-the-workflow-to-push-the-package-to-nugetorg): execute the workflow to push the package to
+NuGet.org
+   - This will trade the GitHub Action OIDC token for a short-lived NuGet API key and perform the package push with the
+     temporary API key.
+
+Let's expand each of these steps for better understanding.
+
+### Prerequisite: set up a GitHub repository for the NuGet package
+
+This step is pretty self explanatory. Many developers, even hobbyist developers who are only producing packages for
+their own consumption, already have their source code stored in a Git repository, hosted on GitHub.
+
+Additionally, GitHub Actions must be enabled for your repository. For public repositories, there is a generous free
```

Comment:
Added! As long as the token is valid and issued from GitHub Actions (the known issuer base URL `https://token.actions.githubusercontent.com`), we will accept it. This can include private repositories.